client/question/services/CurrentQuestionServiceSpec.js

```diff
@@ -90,3 +91,56 @@ describe('CurrentQuestionService', function() {
     });
   });
 });
+
+describe('CurrentQuestionService - Server', function() {
+  var CurrentQuestionService;
+  var $location;
+
+  var $httpBackend;
+  // In the Karma test environment, the deferred promise gets resolved only
+  // when $rootScope.$digest() is called.
+  var $rootScope;
+  beforeEach(module('tie'));
+  var setServerUrl = function(url) {
+    module('tieConfig', function($provide) {
+      $provide.constant('SERVER_URL', url);
+    });
+  };
+  var setDefaultQuestionId = function(id) {
+    module('tieConfig', function($provide) {
+      $provide.constant('DEFAULT_QUESTION_ID', id);
+    });
+  };
+
+  describe('behavior for invalid question - server', function() {
+
+    it('should return null', function(done) {
+      setServerUrl('http://katamari-dam.acy');
+      setDefaultQuestionId(null);
+
+      var PAGE_NOT_FOUND_ERROR = 404;
+      var BAD_REQUEST_ERROR = 404;
+
+      inject(function($injector) {
+        // Set up a non-existent question ID. This must be done before
+        // CurrentQuestionService is initialized.
+        $location = $injector.get('$location');
+        $rootScope = $injector.get('$rootScope');
+        $httpBackend = $injector.get('$httpBackend');
+        $location.search('qid', 'invalidQuestion - server');
+
+        CurrentQuestionService = $injector.get('CurrentQuestionService');
+      });
+      $httpBackend.expectPOST(
```

Comment:
Might want to give some commentary about why two POSTs are expected.

Also, I'm not certain, but don't you need an $httpBackend.flush() at some point? And something like 

```
        $httpBackend.verifyNoOutstandingExpectation();
        $httpBackend.verifyNoOutstandingRequest();
```

Reply:
Definitely! Thanks for the catch.